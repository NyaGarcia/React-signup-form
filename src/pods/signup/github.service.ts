import { GITHUB_ENDPOINT } from 'constants/main.constants';

  export const isGithubAccount = async (username: string): Promise<Boolean> => {
    const response = await fetch(`${GITHUB_ENDPOINT}/${username}`);
    const parsedResponse = await response.json();
    return parsedResponse['login'];
  }
