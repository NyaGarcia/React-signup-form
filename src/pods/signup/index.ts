export { SignupContainer } from "./signup.container";
export { isGithubAccount } from './github.service';
export { validateForm } from './validation.service';