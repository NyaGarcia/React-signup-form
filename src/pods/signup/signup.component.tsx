import React from "react";
import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import CardContent from "@material-ui/core/CardContent";
import { TextField } from "final-form-material-ui";
import Button from "@material-ui/core/Button";
import { Grid } from "@material-ui/core";
import { Form, Field } from "react-final-form";
import { validateForm } from "services/index";


const submit = (values) => {
  alert(JSON.stringify(values));
}
  
const validate = async (values) => {
  const { fieldErrors, recordErrors } = await validateForm(values);
  const errors  = {...getErrorMessages(fieldErrors), ...getErrorMessages(recordErrors)};
  return errors;
  
}

const getErrorMessages = (object): {[field: string]: string} => {
  return Object.keys(object).reduce((acc, key) => {
    const { message } = object[key];
    if(message) acc[key] = message;
    return acc;
  }, {});
  
}

export const SignupComponent = (): JSX.Element => {
 return <Card>
          <CardHeader title="Signup" />
            <CardContent>
              <Form       
                onSubmit={submit}
                validate={validate}
                render={({ handleSubmit, form, pristine, submitting }) => (
                  <form onSubmit={handleSubmit} noValidate>
                    <Grid container alignItems="flex-start" spacing={2}>
                      <Grid item xs={6}>
                        <Field name="id" component={TextField} placeholder="NyaGarcia" label="Github ID" />
                      </Grid>
                      <Grid item xs={7}>
                        <Field name="firstName" component={TextField} placeholder="Nya" label="First Name" />
                      </Grid>
                      <Grid item xs={5}>
                        <Field name="lastName" component={TextField} placeholder="GarcÃ­a" label="Last Name" fullWidth/>
                      </Grid>
                      <Grid item xs={12}>
                        <Field name="email" component={TextField} placeholder="nya@lemoncode.com" label="Email" fullWidth/>
                      </Grid>
                      <Grid item xs={12}>
                        <Field name="repeatEmail" component={TextField} placeholder="nya@lemoncode.com" label="Repeat Email" fullWidth/>
                      </Grid>
                      <Grid item xs={6}>
                        <Field name="password" type="password" component={TextField} placeholder="password" label="Password"/>
                      </Grid>
                      <Grid item xs={6}>
                        <Field name="repeatPassword" type="password" component={TextField} placeholder="password" label="Repeat Password"/>
                      </Grid>
                      <Grid item xs={6}>
                        <Button color="primary" type="submit" variant="contained" disabled={submitting}>Submit</Button>
                      </Grid>
                      <Grid item xs={6}>
                        <Button onClick={form.reset} disabled={pristine}>Reset</Button>
                      </Grid>
                    </Grid>
                  </form>
                )}
              />
            </CardContent>
        </Card>
}