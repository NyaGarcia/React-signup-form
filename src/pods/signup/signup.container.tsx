import React from "react"; 
import { SignupComponent } from "./signup.component";

export const SignupContainer = () => <SignupComponent/>;