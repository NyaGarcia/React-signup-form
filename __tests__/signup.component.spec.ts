import {
  act,
  cleanup,
  fireEvent,
  render,
  waitForElement
} from '@testing-library/react';
import { mount, shallow } from 'enzyme';

import { SignupComponent } from 'pods/signup/signup.component';

const signup = SignupComponent();

describe('Signup component', () => {
  it('Should render correctly', () => {
    const component = shallow(signup);
    expect(component).toMatchSnapshot();
  });

  it('Should capture', async () => {
    const component = mount(signup);
    act(() => {
      const firstName = component
        .find('input')
        .at(0)
        .simulate('change', { target: { target: 'firstName', value: 'Nya' } });
    });

    expect(component.state('firstName')).toEqual('Nya');
  });
});
