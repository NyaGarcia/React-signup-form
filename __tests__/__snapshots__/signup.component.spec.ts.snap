// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component Should render correctly 1`] = `
<ForwardRef(Card)
  classes={
    Object {
      "root": "MuiCard-root",
    }
  }
>
  <WithStyles(ForwardRef(CardHeader))
    title="Signup"
  />
  <WithStyles(ForwardRef(CardContent))>
    <ReactFinalForm
      onSubmit={[Function]}
      render={[Function]}
      validate={[Function]}
    />
  </WithStyles(ForwardRef(CardContent))>
</ForwardRef(Card)>
`;
